### PR TITLE
GSoC 2019 summary report - Jiradet Ounjai

### PR DESCRIPTION
**Pull request by jiradeto**
A summary report for work done during the Google Summer of Code 2019 program.

### CODE REVIEW

File: docs/summer-of-code-ideas/2019-JiradetOunjai.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+## Project Link
+[Improve Shading Language Support in GraphicsFuzz](https://summerofcode.withgoogle.com/projects/#4859963594244096)
+
+
+## Mentors
+Alastair Donaldson, Paul Thomson
+
+
+## Deliverables
```
Comment (afd) [inline docs/summer-of-code-ideas/2019-JiradetOunjai.md]: "providing a new shading language support" sounds like you were supporting a new language altogether; rephrase to talk about making shading language support more complete.

File: docs/summer-of-code-ideas/2019-JiradetOunjai.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+[#555](https://github.com/google/graphicsfuzz/pull/555): Add support for built-in Matrix functions 
+
+[#581](https://github.com/google/graphicsfuzz/pull/581): Refactor geometric builtins 
+
+[#585](https://github.com/google/graphicsfuzz/pull/585): Refactor floating-point pack and unpack built-ins
+
+[#592](https://github.com/google/graphicsfuzz/pull/592): Complete common builtins
+
+
+### Add new ways to generate an opaque expression
```
Comment (afd) [inline docs/summer-of-code-ideas/2019-JiradetOunjai.md]: "Opaque expression, statement uninterpreted and thus non optimizable by compilers" - could be worked on more in terms of English.

File: docs/summer-of-code-ideas/2019-JiradetOunjai.md [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+[#576](https://github.com/google/graphicsfuzz/pull/576): Reducer implementation: reduce global variable declarations to expression
+
+[#622](https://github.com/google/graphicsfuzz/pull/622): Add reducer opportunities
+
+
+### Use GraphicsFuzz to find bugs in other tools
+GraphicsFuzz's team has discovered and reported a bunch of bugs lying in many different GPU vendors and we still keep doing the same as we would be happy if we could help compiler developers detecting more uncovering bugs. Over the past few months, I had a chance to use GraphicsFuzz to help finding bugs in [SPIRV Cross](https://github.com/KhronosGroup/SPIRV-Cross). 
+
+SPIRV-Cross is a very convenient tool that helps parsing and converting SPIR-V to other shader languages. Throughout GSoC program, I relied heavily on this tool since I am using [MoltenVK](https://github.com/KhronosGroup/MoltenVK) to run SPIR-V on Mac and MoltenVK internally calls SPIRV-Cross to convert SPIR-V into Apple's Metal Shading Language.
+
```
Comment (afd) [inline docs/summer-of-code-ideas/2019-JiradetOunjai.md]: Move the "." of "here." outside the link.

File: docs/summer-of-code-ideas/2019-JiradetOunjai.md (hunk 2 above)
Comment (jiradeto) [reply docs/summer-of-code-ideas/2019-JiradetOunjai.md]: Fixed

**Pull request merged**